File: refactored/duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/TestParameterVerifier_testVerifyParametersMissing.cs
public class TestParameterVerifier
{
    [Fact]
    public void TestVerifyParametersMissing1()
    {
        var conf = new Configuration(false);

        var str = "<root xmlns=\"uri:oozie:workflow:0.4\"><parameters>"
                + "<property><name>hello</name></property>"
                + "</parameters></root>";
        try
        {
            ParameterVerifier.VerifyParameters(conf, XmlUtils.ParseXml(str));
        }
        catch (ParameterVerifierException ex)
        {
            Assert.Equal(ErrorCode.E0738, ex.ErrorCode);
            Assert.EndsWith("hello", ex.Message);
            Assert.Contains("1", ex.Message);
            Assert.Equal(0, conf.Size());
        }

        conf = new Configuration(false);

        str = "<root xmlns=\"uri:oozie:workflow:0.4\"><parameters>"
                + "<property><name>hello</name><value>world</value></property>"
                + "</parameters></root>";
        ParameterVerifier.VerifyParameters(conf, XmlUtils.ParseXml(str));
        Assert.Equal(1, conf.Size());
        Assert.Equal("world", conf.Get("hello"));

        conf = new Configuration(false);

        str = "<root xmlns=\"uri:oozie:workflow:0.4\"><parameters>"
                + "<property><name>hello</name></property>"
                + "<property><name>foo</name><value>bar</value></property>"
                + "<property><name>meh</name></property>"
                + "</parameters></root>";
        try
        {
            ParameterVerifier.VerifyParameters(conf, XmlUtils.ParseXml(str));
            Assert.False(true);
        }
        catch (ParameterVerifierException ex)
        {
            Assert.EndsWith("hello, meh", ex.Message);
            Assert.DoesNotContain("foo", ex.Message);
            Assert.Contains("2", ex.Message);
            Assert.Equal("bar", conf.Get("foo"));
        }
    }

    [Fact]
    public void TestVerifyParametersMissing2()
    {
        var conf = new Configuration(false);

        var str = "<root xmlns=\"uri:oozie:workflow:0.4\"><parameters>"
                + "<property><name>hello</name></property>"
                + "</parameters></root>";
        try
        {
            ParameterVerifier.VerifyParameters(conf, XmlUtils.ParseXml(str));
        }
        catch (ParameterVerifierException ex)
        {
            Assert.EndsWith("hello", ex.Message);
            Assert.Contains("1", ex.Message);
            Assert.Equal(0, conf.Size());
        }

        conf = new Configuration(false);

        str = "<root xmlns=\"uri:oozie:workflow:0.4\"><parameters>"
                + "<property><name>hello</name><value>world</value></property>"
                + "</parameters></root>";
        ParameterVerifier.VerifyParameters(conf, XmlUtils.ParseXml(str));
        Assert.Equal("world", conf.Get("hello"));

        conf = new Configuration(false);

        str = "<root xmlns=\"uri:oozie:workflow:0.4\"><parameters>"
                + "<property><name>hello</name></property>"
                + "<property><name>foo</name><value>bar</value></property>"
                + "<property><name>meh</name></property>"
                + "</parameters></root>";
        try
        {
            ParameterVerifier.VerifyParameters(conf, XmlUtils.ParseXml(str));
            Assert.False(true);
        }
        catch (ParameterVerifierException ex)
        {
            Assert.Equal(ErrorCode.E0738, ex.ErrorCode);
            Assert.EndsWith("hello, meh", ex.Message);
            Assert.DoesNotContain("foo", ex.Message);
            Assert.Contains("2", ex.Message);
            Assert.Equal(1, conf.Size());
            Assert.Equal("bar", conf.Get("foo"));
        }
    }
}

File: duplicate-assert/java-junit/TestParameterVerifier_testVerifyParametersMissing.java
public class TestParameterVerifier extends XTestCase {
    public void testVerifyParametersMissing() throws Exception {
        Configuration conf = new Configuration(false);
        
        String str = "<root xmlns=\"uri:oozie:workflow:0.4\"><parameters>"
                + "<property><name>hello</name></property>"
                + "</parameters></root>";
        try {
            ParameterVerifier.verifyParameters(conf, XmlUtils.parseXml(str));
            fail();
        } catch(ParameterVerifierException ex) {
            assertEquals(ErrorCode.E0738, ex.getErrorCode());
            assertTrue(ex.getMessage().endsWith("hello"));
            assertTrue(ex.getMessage().contains("1"));
            assertEquals(0, conf.size());
        }
        
        conf = new Configuration(false);
        
        str = "<root xmlns=\"uri:oozie:workflow:0.4\"><parameters>"
                + "<property><name>hello</name><value>world</value></property>"
                + "</parameters></root>";
        ParameterVerifier.verifyParameters(conf, XmlUtils.parseXml(str));
        assertEquals(1, conf.size());
        assertEquals("world", conf.get("hello"));
        
        conf = new Configuration(false);
        
        str = "<root xmlns=\"uri:oozie:workflow:0.4\"><parameters>"
                + "<property><name>hello</name></property>"
                + "<property><name>foo</name><value>bar</value></property>"
                + "<property><name>meh</name></property>"
                + "</parameters></root>";
        try {
            ParameterVerifier.verifyParameters(conf, XmlUtils.parseXml(str));
            fail();
        } catch(ParameterVerifierException ex) {
            assertEquals(ErrorCode.E0738, ex.getErrorCode());
            assertTrue(ex.getMessage().endsWith("hello, meh"));
            assertFalse(ex.getMessage().contains("foo"));
            assertTrue(ex.getMessage().contains("2"));
            assertEquals(1, conf.size());
            assertEquals("bar", conf.get("foo"));
        }
    }
}
